tests/e2e/utils/adapt_ols_config.py

```diff
@@ -0,0 +1,49 @@
+"""Functionality for updating OLS_CONFIG to support multiproviders"""
+
+import os
+from tests.e2e.utils import cluster
+from tests.e2e.utils import wait_for_ols
+from ols.constants import DEFAULT_CONFIGURATION_FILE
+import yaml
+
+def adapt_ols_config() -> str :
+    """
+        Helper function to adapt to multiple providers during test job by dynamically replacing
+        the existing OLS provider config (olsconfig) with one matching the environment variables during test job execution.
+    """
+    try:
+
+        provider = os.environ["PROVIDER"]
+        model = os.environ["MODEL"]
+        key_path = os.environ["PROVIDER_KEY_PATH"]
+        provider_url = os.environ["OLS_PROVIDER_URL"]
+        configmap_yaml = cluster.run_oc(["get", "cm/olsconfig", "-o", "yaml"]).stdout
+        configmap = yaml.safe_load(configmap_yaml)
+        olsconfig = yaml.safe_load(configmap["data"][DEFAULT_CONFIGURATION_FILE])
+
+        olsconfig["ols_config"]["logging_config"]["lib_log_level"] = "INFO"
+
+        olsconfig["ols_config"]["providers"] = [{
+            "name": provider,
+            "type": provider,
+            "url": provider_url,
+            "credentials_path": key_path,
+            "models": [{"name": model}]
+        }]
+
+        configmap["data"][DEFAULT_CONFIGURATION_FILE] = yaml.dump(olsconfig)
+        updated_configmap = yaml.dump(configmap)
+
+        cluster.run_oc(["delete", "configmap", "olsconfig"])
+        cluster.run_oc(["apply", "-f", "-"], command=updated_configmap)
```

Comment:
Instead of overriting the configmap, I'd say it would make sense to modify the olsconfig Custom Resource. It's what we do in the ols_installer during installation https://github.com/openshift/lightspeed-service/blob/main/tests/e2e/utils/ols_installer.py#L346-L380
Essentially you pick the olsconfig from the list based on the provider + any extra options (like introspection) and just apply the CR.